include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -410,10 +410,21 @@ class transform_iterator
     typedef typename ::std::iterator_traits<_Iter>::pointer pointer;
     typedef typename ::std::iterator_traits<_Iter>::iterator_category iterator_category;
 
-    transform_iterator(_Iter __it = _Iter(), _UnaryFunc __unary_func = _UnaryFunc())
-        : __my_it_(__it), __my_unary_func_(__unary_func)
+    //only enable this constructor if both template types are default constructible
+    template <typename _Iter_ = _Iter, typename _UnaryFunc_ = _UnaryFunc,
+              typename = ::std::enable_if_t<
+                  ::std::is_default_constructible_v<_Iter_>&& ::std::is_default_constructible_v<_UnaryFunc_>>>
+    transform_iterator() : __my_it_(), __my_unary_func_()
     {
     }
+    //only enable this constructor if the unary functor is default constructible
+    template <typename _UnaryFunc_ = _UnaryFunc,
```

Comment:
Do we want to have less similar name of Dummy template parameter? It's hard to mess it with the original one and check the wrong one in `enable_if` condition?

Reply:
I've improved it.